.github/workflows/release_ruleset.yaml

```diff
@@ -2,28 +2,47 @@
 # See example usage in https://github.com/bazel-contrib/rules-template/blob/main/.github/workflows/release.yaml
 #
 # By default this workflows calls `.github/workflows/release_prep.sh` as the command to prepare
-# the release. This can be customized with the `release_prep_command` attribute. Release notes are
-# expected to be outputted to stdout from the release prep command.
+# the release. This can be customized with the `release_prep_command` attribute. See documentation
+# for the `release_prep_command` property below for inputs and outputs of the script.
 #
 # This workflow uses https://github.com/bazel-contrib/setup-bazel to prepare the cache folders.
 # Caching may be disabled by setting `mount_bazel_caches` to false.
+#
+# The workflow requires the following permissions to be set on the invoking job:
+#
+# permissions:
+#   contents: write        # Needed for uploading release files
+
 
 on:
   # Make this workflow reusable, see
   # https://github.blog/2022-02-10-using-reusable-workflows-github-actions
   workflow_call:
     inputs:
-      release_files:
-        required: true
-        description: |
-          Newline-delimited globs of paths to assets to upload for release.
-          See https://github.com/softprops/action-gh-release#inputs
-        type: string
       release_prep_command:
         default: .github/workflows/release_prep.sh
         description: |
           Command to run to prepare the release and generate release notes.
-          Release notes are expected to be outputted to stdout.
+
+          The script will be provided with the following env vars:
+
+            ARTIFACTS_DIR: Path to a pre-created directory where all artifacts to be
+            uploaded to the GitHub release must be placed.
+
+            RELEASE_NOTES: Path to a pre-created file where release notes content should
+            be written.
+
+          The following arguments are passed to the script:
+
+            <tag_or_ref> The tag supplied as an input to this workflows, or ${{ github.ref_name }}.
+
+          The script must output a JSON blob with paths to release archives. The path can
+          be absolute or relative to the ARTIFACTS_DIR. For example:
+
+          {
+            "release_archives": ["my-ruleset-vX.Y.Z.tar.gz"]
+          }
```

Comment:
I can understand this new API for release_prep being controversial because it will be a breaking change. I'm open to ideas to try to make it less breaking. The crux is that we need a way to identify release archives for the attestations later on in the workflow.

Reply:
I chose JSON output so that it's extensible in the future, for example, maybe we'll have a `attest_artifacts` property for users who wish to attest more files. GitHub runners have `jq` installed so parsing is easy.